app/services/data_requests/lookup_shared_device_users.rb

```diff
@@ -9,66 +9,26 @@ module DataRequests
   # network are found.
   #
   class LookupSharedDeviceUsers
-    attr_reader :initial_users
+    attr_reader :initial_users, :depth
 
-    def initialize(initial_users)
+    def initialize(initial_users, depth = 3)
       @initial_users = initial_users
+      @depth = depth
+      @user_ids = initial_users.map(&:id)
+      @device_cookie_uuids = []
```

Comment:
let's make these sets?
```suggestion
      @user_ids = initial_users.map(&:id).to_set
      @device_cookie_uuids = Set.new
```